api/src/test/java/com/epam/pipeline/controller/region/CloudRegionControllerTest.java

```diff
@@ -0,0 +1,514 @@
+        Mockito.doReturn(awsRegion).when(mockCloudRegionApiService).create(Mockito.refEq(awsRegionDTO));
+
+        final MvcResult mvcResult = mvc().perform(post(REGION_URL)
+                .servletPath(SERVLET_PATH)
+                .contentType(EXPECTED_CONTENT_TYPE)
+                .content(getObjectMapper().writeValueAsString(awsRegionDTO)))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(EXPECTED_CONTENT_TYPE))
+                .andReturn();
+
+        final ArgumentCaptor<AWSRegionDTO> awsRegionDTOCaptor = ArgumentCaptor.forClass(AWSRegionDTO.class);
+        Mockito.verify(mockCloudRegionApiService).create(awsRegionDTOCaptor.capture());
+        Assertions.assertThat(awsRegionDTOCaptor.getValue()).isEqualToComparingFieldByField(awsRegionDTO);
+
+        Mockito.verify(mockCloudRegionApiService).create(Mockito.refEq(awsRegionDTO));
+
+        ControllerTestUtils.assertResponse(mvcResult, getObjectMapper(), expectedAwsRegionResult,
+                new TypeReference<Result<AwsRegion>>() { });
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldCreateAzureRegion() throws Exception {
+        Mockito.doReturn(azureRegion).when(mockCloudRegionApiService).create(Mockito.refEq(azureRegionDTO));
+
+        final MvcResult mvcResult = mvc().perform(post(REGION_URL)
+                .servletPath(SERVLET_PATH)
+                .contentType(EXPECTED_CONTENT_TYPE)
+                .content(getObjectMapper().writeValueAsString(azureRegionDTO)))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(EXPECTED_CONTENT_TYPE))
+                .andReturn();
+
+        final ArgumentCaptor<AzureRegionDTO> azureRegionDTOCaptor = ArgumentCaptor.forClass(AzureRegionDTO.class);
+        Mockito.verify(mockCloudRegionApiService).create(azureRegionDTOCaptor.capture());
+        Assertions.assertThat(azureRegionDTOCaptor.getValue().getAuthFile())
+                .isEqualTo(azureRegionDTO.getAuthFile());
+
+        Mockito.verify(mockCloudRegionApiService).create(Mockito.refEq(azureRegionDTO));
+
+        ControllerTestUtils.assertResponse(mvcResult, getObjectMapper(), expectedAzureResult,
+                new TypeReference<Result<AzureRegion>>() { });
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldCreateGcpRegion() throws Exception {
+        Mockito.doReturn(gcpRegion).when(mockCloudRegionApiService).create(Mockito.refEq(gcpRegionDTO));
+
+        final MvcResult mvcResult = mvc().perform(post(REGION_URL)
+                .servletPath(SERVLET_PATH)
+                .contentType(EXPECTED_CONTENT_TYPE)
+                .content(getObjectMapper().writeValueAsString(gcpRegionDTO)))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(EXPECTED_CONTENT_TYPE))
+                .andReturn();
+
+        final ArgumentCaptor<GCPRegionDTO> gcpRegionDTOCaptor = ArgumentCaptor.forClass(GCPRegionDTO.class);
```

Comment:
In general, I see no point to use argument captor here and elsewhere in this class. But if for some reasons it necessary let's discuss it.